frontend/src/app/AdminNavBar/AdminNavBar.tsx

```diff
@@ -78,13 +73,44 @@ export interface AdminNavBarProps {
 }
 
 export const AdminNavBar = ({ isMenuOpen }: AdminNavBarProps): JSX.Element => {
-  const { user } = useUser()
+  const { user, isLoading: isUserLoading } = useUser()
 
-  const {
-    isOpen: isContactModalOpen,
-    onClose: onContactModalClose,
-    onOpen: onContactModalOpen,
-  } = useDisclosure()
+  const emergencyContactKey = useMemo(
+    () =>
+      user?._id
+        ? user.contact
+          ? null
+          : EMERGENCY_CONTACT_KEY_PREFIX + user._id
+        : null,
+    [user],
+  )
+
+  const [hasSeenContactModal, setHasSeenContactModal] =
+    useLocalStorage<boolean>(emergencyContactKey)
+
+  const [isContactModalOpen, setIsContactModalOpen] = useState<boolean>(false)
+  const handleContactModalOpenClick = () => {
+    setIsContactModalOpen(true)
+  }
+  const onEmergencyContactModalClose = () => {
+    setIsContactModalOpen(false)
+    setHasSeenContactModal(true)
+  }
+
+  const isEmergencyContactModalOpenOnLogin = useMemo(
+    () => !isUserLoading && !user?.contact && !hasSeenContactModal,
+    [isUserLoading, hasSeenContactModal, user],
+  )
+
+  const isEmergencyContactModalOpen =
+    (isContactModalOpen && Boolean(user)) || isEmergencyContactModalOpenOnLogin
```

Comment:
if we want to open modal when something has loaded, we could use `useEffect` to trigger it instead.

```tsx
useEffect(() => {
  if (emergencyContactKey) {
    onOpen() // from useDisclosure
  }
}, [])
```